testing/jobs.pr.cloudbuild.yaml

```diff
@@ -0,0 +1,92 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+
+- id: 'Lint'
+  name: 'golang:1.18-alpine'
+  dir: '${_SAMPLE_DIR}'
+  entrypoint: /bin/sh
+  args: 
+  - '-c'
+  - |
+    go install golang.org/x/tools/cmd/goimports@latest
+    if [ -n "$(goimports -d .)" ]; then
+      echo 'To fix this check, run "goimports -w ."'
+      exit 1
+    fi
+
+- id: 'Unit Tests'
+  name: 'golang:1.18-alpine'
+  dir: '${_SAMPLE_DIR}'
+  args: ['test', '-v']
+
+- id: 'Build Container Image'
+  name: 'gcr.io/cloud-builders/gcloud:$_CLOUDSDK_VERSION'
+  dir: '${_SAMPLE_DIR}'
+  args: ['builds', 'submit', '--pack', 'image=gcr.io/${PROJECT_ID}/${_IMAGE}:${SHORT_SHA}'] # Tag docker image with git commit sha
+
+- id: 'Deploy to Cloud Run'
+  name: 'gcr.io/cloud-builders/gcloud:$_CLOUDSDK_VERSION'
+  entrypoint: /bin/bash
+  args:
+  - '-c'
+  - |
+    gcloud components install alpha --quiet
+    
+    gcloud alpha run jobs create ${_IMAGE}-${SHORT_SHA} \
+      --image gcr.io/${PROJECT_ID}/${_IMAGE}:${SHORT_SHA} \
+      --project ${PROJECT_ID} \
+      --region ${_REGION} \
+      --tasks 2 \
+      --set-env-vars=SLEEP_MS=1000 \
+      --max-retries 0 \
+      --wait
+
+- id: 'Integration Tests'
```

Comment:
I might rename this "Check Job Logs" or something, because it is checking on results, not actually running tests.